### DISCUSSION TITLE
Fixed / Pinned nodes with a layout running

### DISCUSSION THREAD

**Discussion by xShields** [graphology/graphology#375] [Q&A] [answered]
Hi,

I was trying to figure out how to do "fixed" nodes with graphology / sigma, and thought this would be the best way to ask about it. Specifically, I'm trying to replicate the feature specified in Vis JS, the fixed option: [https://visjs.github.io/vis-network/docs/network/nodes.html](https://visjs.github.io/vis-network/docs/network/nodes.html)

They define it as: "When true, the node will not move but IS part of the physics simulation. When defined as an object, movement in either X or Y direction can be disabled."

This is extremely helpful interacting with graphs. My use case is letting a user drag a node, and then automatically "pinning" it, setting it to fixed, when they let go. In VisJS, if the simulation is still running, it continues to organize the non-fixed nodes neatly around whatever the user has moved around. Also helps in other situations, when the layout doesn't quite get things perfectly spaced, for example. 

Right now, I have drag and drop working, but once the simulation continues, it snaps those nodes I dragged back to where it wants to place them.

I saw these issue:

[https://github.com/graphology/graphology/issues/207](https://github.com/graphology/graphology/issues/207)
[https://github.com/graphology/graphology/issues/324](https://github.com/graphology/graphology/issues/324)

I wasn't sure if this implied there is no way to do node fixing / pinning at the moment, or just that it needs to be improved upon. Is this something that's possible to implement right now, even if it's not a clean solution? If not, is there any ETA on completion? I understand this is open source, so appreciate all the work you guys have put into this regardless!

Thanks for your time

**Comment by Yomguithereal** [comment, marked as answer]
I am assuming you are speaking of the ForceAtlas2 layout, in which case you just need to have a node attribute named `fixed` set to `true` for this to work. #207 is about completely ignoring nodes which is different and #324 is about harmonizing the way FA2 works with other layout algorithms such as noverlap API-wise.

**Reply by xShields** [reply]
Great, thank you! Correct, was referring to ForceAtlas2. Was there somewhere in the documentation that lays out these properties? If not, is there somewhere in the code I should be looking when I have similar questions on functionality / behavior?

**Reply by Yomguithereal** [reply]
This is not really documented no sorry, because it will probably evolve for harmonization reasons. You can check the related code here: https://github.com/graphology/graphology/blob/master/src/layout-forceatlas2/helpers.js#L132-L142 and then the iteration code itself.
